### ISSUE TITLE
Updating from 0.24.6->0.25.21 somewhere causes name clash between preAggregations and measures

### ISSUE THREAD

**Issue by benswinburne** [cube-js/cube.js#1825]
**Describe the bug**
I was running 0.24.6, and upgraded in order to run a release containing the [fix for SELECT 1][1] but in doing so there appears to be an undocumented breaking change.

```
cube_1   | Error: Compile errors:
cube_1   | Events cube: count defined more than once
cube_1   |     at ErrorReporter.throwIfAny (/cube/node_modules/@cubejs-backend/schema-compiler/src/compiler/DataSchemaCompiler.js:46:13)
cube_1   |     at DataSchemaCompiler.throwIfAnyErrors (/cube/node_modules/@cubejs-backend/schema-compiler/src/compiler/DataSchemaCompiler.js:182:23)
cube_1   |     at /cube/node_modules/@cubejs-backend/schema-compiler/src/compiler/DataSchemaCompiler.js:106:16
cube_1   | Internal Server Error: dbc48cd8-5683-4a3c-892f-5b6d841af635-span-1 (22ms)
```

**To Reproduce**
1. Install 0.24.6
2. Create a measure entitled `count` and a preaggregation entitled `count`.
3. Run server, observe no errors
4. Update to 0.25.21
5. Run server, observe error

**Expected behavior**
I'd expect the names of two unrelated parts of the configuration to not conflict.

I'd expect a breaking change if it was intentional to be listed in the changelog, but it's more likely a bug introduced somewhere between these two versions which is more understandable as to why it is not listed.

**Version:**
0.24.6 -> 0.25.21

[1]: https://github.com/cube-js/cube.js/pull/1786

**Comment by ovr** [comment]
Hello @benswinburne,

This check was added in 2018, 2 years ago

https://github.com/cube-js/cube.js/blame/6a21aa5a2e2323ab10cdae73a4f1b3450e4be74f/packages/cubejs-schema-compiler/src/compiler/CubeSymbols.js#L94

**Comment by benswinburne** [comment]
Presumably it wasn't working in 0.24.6 then?

I've been running a Cube 0.24.6 in production with a preAggregation and a measure named `count` for a while now with no issues 😬

**Comment by ovr** [comment]
Sorry for the late reply, I rechecked the source code, and I found what was changed.

https://github.com/cube-js/cube.js/commit/bffd22095f58369f3d52474283951b4844657f2b#diff-15f938408d0ce33aabdb92218d1b85cf8720b9ba217ccceb7b96f11b94ad5d81R91

On the first look, it's a breaking change, but it's possible to reference dimensions/measure/pre-aggs inside the Cube.js schema file. It's why it started to check it as expected to protect unexpected behaviors.

Thanks
